pytests/stubs/pyclasses.pyi

```diff
@@ -1,5 +1,31 @@
-class AssertingBaseClass: ...
+class AssertingBaseClass:
+    def __new__(cls, /, expected_type): ...
+
+class ClassWithDecorators:
+    def __new__(cls, /): ...
+    @property
+    def attr(self, /): ...
+    @attr.setter
+    def attr(self, /, value): ...
+    @classmethod
+    @property
+    def cls_attribute(cls, /): ...
+    @classmethod
+    def cls_method(cls, /): ...
+    @staticmethod
+    def static_method(): ...
+
 class ClassWithoutConstructor: ...
-class EmptyClass: ...
-class PyClassIter: ...
-class PyClassThreadIter: ...
+
+class EmptyClass:
+    def __len__(self, /): ...
+    def __new__(cls, /): ...
+    def method(self, /): ...
```

Comment:
Future extension: ability to pull docstrings out of the Rust code and onto these stubs :)